eos-updater-avahi/eos-updater-avahi.c

```diff
@@ -78,9 +78,20 @@ get_raw_summary_timestamp_from_metadata (GBytes    *summary,
                                          GError   **error)
 {
   g_autoptr(GVariant) summary_variant = g_variant_ref_sink (g_variant_new_from_bytes (OSTREE_SUMMARY_GVARIANT_FORMAT, summary, FALSE));
+  g_autoptr(GVariant) additional_metadata = NULL;
   g_autoptr(GVariant) value = NULL;
 
-  value = g_variant_lookup_value (summary_variant, "ostree.summary.last-modified", G_VARIANT_TYPE_UINT64);
+  if (!g_variant_is_normal_form (summary_variant))
+    {
+      *out_found = FALSE;
+      *out_timestamp = 0;
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
+                   "Corrupt summary file");
```

Comment:
Is `G_IO_ERROR_NOT_FOUND` the best error code for this? I've noticed that callers will often check `NOT_FOUND` as an indicator that some file needs to be created. Perhaps `G_IO_ERROR_FAILED` (or our own error code)?

Reply:
Callers should do whatever’s documented for the method they’re calling. I don’t think we’re returning `G_IO_ERROR_NOT_FOUND` from elsewhere in this function, so it’s distinct. I’ll have a think about whether another code is more appropriate.

Reply:
I think the main problem is that its easy for `G_IO_ERROR_NOT_FOUND` to collide:

    gboolean some_func (..., GError **error)
      {
        if (!g_file_get_contents ("some-file", &contents, &error))
          return FALSE;

        if (!get_raw_summary_timestamp_from_metadata(..., &error))
          return FALSE;
      }

Callers of `some_func` might respond to `G_IO_ERROR_NOT_FOUND` by creating the file requested by `g_file_get_contents` and trying again (even though the issue was actually with`get_raw_summary_timestamp_from_metadata`)

Reply:
Yeah, I’m satisfied that `G_IO_ERROR_NOT_FOUND` is appropriate here. If a caller wants to create the ‘missing’ file, they’ll call `ostree_repo_regenerate_summary()` which doesn’t care whether the `summary` file exists already.